### PR TITLE
Upgrade to React 16, cleanup core deps

### PR DESCRIPTION
**Pull request by balthazar**
I think the only issue I noticed was with the jsx-layers. Need to investigate if the problem is also present on master.

Might be interesting to discuss whenever we should add an option on the react wrapper to catch errors down the chain.

Related to #917

### CODE REVIEW

**Comment by Pessimistress** [comment]
Should remove all the string `ref`s https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "build": "npm run build-clean && npm run build-es6 && npm run build-es5",
     "cover": "NODE_ENV=test nyc --reporter html --reporter cobertura --reporter=lcov npm run test-cover",
     "lint": "eslint src test examples && npm run lint-yarn",
-    "lint-yarn": "!(grep -q unpm.u yarn.lock) || (echo 'Please rebuild yarn file using public npmrc' && false)",
+    "lint-yarn": "!(grep -q unpm.u yarn.lock) && echo 'Lockfile valid.' || (echo 'Please rebuild yarn file using public npmrc' && false)",
```
Comment (ibgreen) [inline package.json]: One more parenthesis?

File: package.json [hunk 2]
```diff
@@ -78,15 +76,14 @@
     "pre-commit": "^1.2.2",
     "probe.gl": "^0.0.2",
     "raw-loader": "^0.5.1",
-    "react": "^15.4.0",
-    "react-dom": "^15.4.0",
-    "reify": "^0.4.4",
+    "react": "^16.2.0",
+    "react-dom": "^16.2.0",
+    "reify": "0.4.4",
```
Comment (ibgreen) [inline package.json]: Does reify need to be pinned now?

Comment (balthazar) [reply package.json]: Trying to update to the latest breaks the build, so yes. I preferred being explicit by pinning it.

File: package.json (hunk 1 above)
Comment (balthazar) [reply package.json]: not sure, why? Is that required?

**Comment by balthazar** [comment]
@Pessimistress updated the refs I could find, is there some you were thinking that I missed?

**Pull request merged**